src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -762,6 +762,39 @@ static void uct_ud_mlx5_qp_update_caps(struct ibv_qp_cap *qp_caps)
     qp_caps->max_inline_data = 0;
 }
 
+int uct_ud_mlx5_ep_is_connected(const uct_ep_h tl_ep,
+                                const uct_ep_is_connected_params_t *params)
+{
+    uct_ud_mlx5_ep_t *ep = ucs_derived_of(tl_ep, uct_ud_mlx5_ep_t);
+    uct_ib_address_t *ib_addr;
+    uct_ud_iface_addr_t *iface_addr;
+    const uct_ud_ep_addr_t *ep_addr;
+    uint32_t qp_num;
+    uint8_t *rgid;
+
+    UCT_EP_IS_CONNECTED_CHECK_DEV_IFACE_ADDRS(params);
+
+    if (params->field_mask & UCT_EP_IS_CONNECTED_FIELD_EP_ADDR) {
+        ep_addr = (const uct_ud_ep_addr_t*)params->ep_addr;
+        if (ep->super.dest_ep_id != uct_ib_unpack_uint24(ep_addr->ep_id)) {
+            return 0;
+        }
+    }
+
+    ib_addr    = (uct_ib_address_t*)params->device_addr;
+    iface_addr = (uct_ud_iface_addr_t*)params->iface_addr;
+
+    rgid = NULL;
+    if (ep->peer_address.is_global) {
+        rgid = ep->peer_address.grh_av.rgid;
+    }
+
+    qp_num = uct_ib_unpack_uint24(iface_addr->qp_num);
+    return uct_ib_iface_is_same_device(ib_addr, ep->peer_address.av.rlid >> 8,
+                                       (union ibv_gid*)rgid) &&
+           (ep->peer_address.av.dqp_dct == qp_num);
```

Comment:
shall we convert dqp_dct to host endianness?
how did it work?

Reply:
worked because I shifted 8 bits right, but indeed should convert.

Reply:
the `ep->peer_address.av.dqp_dct` test is added only here, probably because I changed it few days ago, so it wasn't checked by the reconf PR.

Reply:
Each PR should be tested individually. This means need to add gtests to this PR. it could be UCT level tests that connect to some address and the checks is_connected returns true. And that is_connected returns false when getting a different entity's address.